api/client/src/main/java/org/projectnessie/client/rest/ResponseCheckFilter.java

```diff
@@ -84,52 +83,85 @@ public static void checkResponse(ResponseContext con) throws Exception {
         exception = new NessieServiceException(error);
     }
 
-    if (error.getClientProcessingException() != null) {
-      exception.addSuppressed(error.getClientProcessingException()); // for trouble-shooting
```

Comment:
Why not include it as an exception using JRE APIs? Having full text in the `error` is good, but it might be valuable to track the exception itself too to be formatted according to the logger settings (which may not be plain text).